tests/integ/scala/test_file_operation_suite.py

```diff
@@ -219,6 +222,55 @@ def test_put_negative(session, temp_stage, temp_source_directory, path1):
     assert "does not exist or not authorized." in str(stage_not_exist_info)
 
 
+def test_put_stream_with_one_file(session, temp_stage, path1, path2, path3):
+    stage_prefix = f"prefix_{random_alphanumeric_name()}"
+    stage_with_prefix = f"@{temp_stage}/{stage_prefix}"
+    file_name = os.path.basename(path1)
+    with open(path1, "rb") as fd:
+        first_result = session.file.put_stream(fd, stage_with_prefix, file_name)
+    assert first_result.source == os.path.basename(path1)
+    assert first_result.target == os.path.basename(path1) + ".gz"
+    assert first_result.source_size in (10, 11)
+    assert first_result.target_size in (32, 64)
+    assert first_result.source_compression == "NONE"
+    assert first_result.target_compression == "GZIP"
+    assert first_result.status == "UPLOADED"
+    assert first_result.message == ""
+
+
+def test_put_stream_withone_file_twice(session, temp_stage, path1):
+    stage_prefix = f"prefix_{random_alphanumeric_name()}"
+    stage_with_prefix = f"@{temp_stage}/{stage_prefix}"
+    file_name = os.path.basename(path1)
+    fd = open(path1, "rb")
+    session.file.put_stream(fd, stage_with_prefix, file_name)
+
+    # put same file again
+    second_result = session.file.put_stream(
+        fd, stage_with_prefix, file_name, overwrite=False
+    )
+    assert second_result.source == os.path.basename(path1)
+    assert second_result.target == os.path.basename(path1) + ".gz"
+    assert second_result.source_size in (10, 11)
+    assert second_result.target_size in (0, 32)
+    assert second_result.source_compression == "NONE"
+    assert second_result.target_compression == "GZIP"
+    assert second_result.status in ("SKIPPED", "UPLOADED")
+    assert second_result.message == ""
+
+
+def test_put_stream_negative(session, temp_stage, temp_source_directory, path1):
+    stage_prefix = f"prefix_{random_alphanumeric_name()}"
+    stage_with_prefix = f"@{temp_stage}/{stage_prefix}"
+    file_name = os.path.basename(path1)
+    fd = open(path1, "rb")
+    fd.close()
+
+    with pytest.raises(SnowparkUploadUdfFileException) as ex_info:
```

Comment:
we should have a new exception because now this API is not used for udf only.

Reply:
I talked to Jianzhun about this offline and it seems like a new exception for this could mean another behavior change. Should I proceed with BCR?

Reply:
To avoid BCR, probably we can check whether it’s used in UDF, if it is, then throw the original error; if not, throw a new error. @sfc-gh-yixie wdyt?

Reply:
We can create a new upload exception class for `put_stream`. To avoid break change, UDF still throws a SnowparkUploadUdfFileException, which can be changed to be a subclass of the new upload exception.